device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -878,6 +878,8 @@ public void unregisterMultiplexedDeviceClient(List<ClientConfiguration> configs,
                         throw new MultiplexingClientDeviceRegistrationTimeoutException("Timed out waiting for all device unregistrations to finish.");
                     }
                 }
+
+                this.deviceConnectionStates.remove(newlyUnregisteredConfig.getDeviceId());
```

Comment:
This was missing because otherwise the last connection state of a newly unregistered device would stick around indefinitely. Not the cause of the broader memory leak, though.

Reply:
Any chance this call could fail?

Reply:
Or threading issues with a device leaving multiplexing and then being added right away?

Reply:
We have some synchronization in place at the multiplexing client level that prevents simultaneous registration and unregistration, so we are safe in that regard.

Reply:
> Any chance this call could fail?

No. If the key isn't present in the map, the method simply returns null. We don't care about the returned object here regardless.